examples/distribute-across-az/init.pp

```diff
@@ -0,0 +1,71 @@
+# Feel free to change these variables to launch a different number
+# of instances or launch them in different availability zones
+$availability_zones = ['sa-east-1a', 'sa-east-1b']
+$number_of_app_servers = "5"
+$maximum_app_servers = 10
+$region = 'sa-east-1'
+$ami = 'ami-67a60d7a'
+
+# Don't change below this line
+# First lets validate the variables passed in
+assert_type(Array[String[1]], $availability_zones)
```

Comment:
The whole lower part could be wrapped into a define. Then the `assert_type` calls here could be replaced by type annotations on the parameters of the define.